### ISSUE TITLE
malicious xz version

### ISSUE THREAD

**Issue by reecer** [msys2/MSYS2-packages#4474]
The xz version (5.6.1) being used at [xz/PKGBUILD](https://github.com/msys2/MSYS2-packages/blob/master/xz/PKGBUILD) contains a backdoor: https://www.openwall.com/lists/oss-security/2024/03/29/4

I think it should be reverted to (5.4.6) like it was previously: https://github.com/msys2/MSYS2-packages/blob/d7c37ffdadd8513f84a16b1565a5f9df75e37af2/xz/PKGBUILD

**Comment by Biswa96** [comment]
Upstream issue https://github.com/tukaani-project/xz/issues/92

**Comment by lazka** [comment]
I went with what Arch Linux did for now, and went with the git checkout: #4475

**Comment by lazka** [comment]
btw, the last pacman was built against 5.4.5-1 (relevant since it statically links everything)

**Comment by yumetodo** [comment]
https://github.com/tukaani-project/xz

It seems to me that xz repo was blocked by github.

![image](https://github.com/msys2/MSYS2-packages/assets/10869046/20ca1b2d-91a8-4c90-b54c-e5b79d3f2b53)

**Comment by Biswa96** [comment]
Would it be possible to use Debian repository https://salsa.debian.org/debian/xz-utils ? I am not sure if it includes some changes specific to Debian.

**Comment by jeremyd2019** [comment]
It looks like gentoo went back to 5.4.2, which was the last release signed by the other developer.  But there is also some discussion whether they should revert all the way back to a 5.2 release that contains no commits by that developer.  Of course, not having the repo available to look at makes it rather harder to do that analysis.  Have to use something like that debian fork.  They do at least have "upstream" tags published, so you can get versions without the debian patches from there if necessary.

**Comment by MehdiChinoune** [comment]
@jeremyd2019 you could browse the repo in https://git.tukaani.org/?p=xz.git

**Comment by Kreijstal** [comment]
> @jeremyd2019 you could browse the repo in https://git.tukaani.org/?p=xz.git

is that domain to be trusted

**Comment by Biswa96** [comment]
> is that domain to be trusted

From https://tukaani.org/xz-backdoor/
> Only I have had access to the main tukaani.org website, git.tukaani.org repositories, and related files.

**Comment by jeremyd2019** [comment]
I don't know - at this point I think it's still kind of an open question of *who* or *what* to trust 😱.

**Comment by sh-shahrokhi** [comment]
Please revert to 5.2 or at least 5.4 as soon as possible

**Comment by MehdiChinoune** [comment]
> Please revert to 5.2 or at least 5.4 as soon as possible

Why?
You should read more about the backdoor before commenting.

**Comment by pfusik** [comment]
Homebrew reverted to 5.4.6.

**Comment by Biswa96** [comment]
ArchLinux moved to tukaani-dot-org git repository https://gitlab.archlinux.org/archlinux/packaging/packages/xz/-/blob/main/PKGBUILD?ref_type=heads

AlpineLinux moved to GitHub generated tarball https://git.alpinelinux.org/aports/tree/main/xz/APKBUILD

**Comment by stevenclarke9** [comment]
The website https://tukaani.org/xz-backdoor/ is a brief account of what is happening with the github repo.
A lot of the tech websites including wired have written articles on this.
The hack wasnt the work of one author it was done by an apt based in Russia. Some of the commit dates did not have the timezone UTC+8 of the singapore vpn they used. Feb 7 2022 is the first date this JiaT75 submitted a patch to XZ Utils repo.
The domain is controlled by the orginal author his name is Lasse Collin. The next major version will likely be 5.8

**Comment by stevenclarke9** [comment]
He has got a big job to repatch to fix the real bugs starting from the commit on 12 July 2022.

**Comment by MehdiChinoune** [comment]
> it was done by an apt based in Russia.

Any proof of that, that contributor started his work since Oct, 2021 (before war)

**Comment by lazka** [comment]
I don't think this a the right place for speculations. Please try to keep things on topic: xz things relating to the MSYS2 package
